db-builder/src/main/java/com/yahoo/covid19/database/models/Places.java

```diff
@@ -85,126 +86,76 @@ private PreparedStatement getRelationshipStatement(DatabaseBuilder.DBConnector c
     @Override
     public List<PreparedStatement> getStatement(DatabaseBuilder.DBConnector connector) throws SQLException {
         List<PreparedStatement> statements = new ArrayList<>();
-        String parentId;
-        switch (type) {
-            case Supername:
-                parentId = null;
-                break;
-            case Country:
-                parentId = EARTH_ID;
-                break;
-            case StateAdminArea:
-                parentId = countryId;
-                break;
-            case CountyAdminArea:
-                parentId = stateId;
-                break;
-            default:
-                throw new IllegalStateException("invalid geographical type: " + type);
-        }
         statements.add(getCommonStatement(connector));
-        if (parentId != null) {
-            statements.add(getRelationshipStatement(connector, parentId));
+
+        for (String parent : parentId) {
+            statements.add(getRelationshipStatement(connector, parent));
         }
 
         return statements;
     }
 
     @Override
     public boolean isValid(Map<String, Insertable> foreignKeyMap) {
-        if (id == null || id.isEmpty() || type == null) {
-            if (type == null) {
-                errorCode = INVALID_TYPE;
-            } else {
-                errorCode = INVALID_ID;
-            }
-            return false;
-        }
-
-        switch (type) {
-            case StateAdminArea:
-                return validateState(foreignKeyMap);
-            case CountyAdminArea:
-                return validateCounty(foreignKeyMap);
-            case Country:
-                return validateCountry(foreignKeyMap);
-            case Supername:
-                return validateEarth(foreignKeyMap);
-            default:
-                return false;
+        if (isValid != null) {
+            return isValid;
         }
+        isValid = isValidRegion(foreignKeyMap);
+        return isValid;
     }
 
-    public boolean isValidCoordinate(String coordinate) {
-        if (coordinate == null) {
-            errorCode = INVALID_COORDINATE;
-            return false;
-        }
-
-        try {
-            Double.valueOf(coordinate);
-        } catch (NumberFormatException e) {
-            errorCode = INVALID_COORDINATE;
+    private boolean isValidRegion(Map<String, Insertable> foreignKeyMap) {
+        if (id == null || id.isEmpty()) {
+            errorCode = INVALID_ID;
             return false;
         }
-        return true;
-    }
 
-    public boolean validateState(Map<String, Insertable> foreignKeyMap) {
-        if (! isValidCoordinate(latitude) || ! isValidCoordinate(longitude)) {
-            errorCode = INVALID_COORDINATE;
+        if (type == null) {
```

Comment:
I'd remove the isValidRegion function and just move all this logic back into isValid.